src/Elliptic/DiscontinuousGalerkin/InitializeFluxes.hpp

```diff
@@ -30,45 +34,50 @@ namespace Actions {
 /*!
  * \brief Initialize DataBox tags related to discontinuous Galerkin fluxes
  *
- * With:
- * - `normal_dot_flux_tag` = `db::add_tag_prefix<Tags::NormalDotFlux,
- * variables_tag>`
- * - `interface<Tag>` =
- * `Tags::Interface<Tags::InternalDirections<volume_dim>, Tag>`
- * - `boundary<Tag>` =
- * `Tags::Interface<Tags::BoundaryDirectionsInterior<volume_dim>, Tag>`
- *
  * Uses:
- * - Metavariables:
- *   - Items required by `flux_comm_types`
  * - System:
  *   - `volume_dim`
  *   - `variables_tag`
- * - DataBox:
- *   - `Tags::InternalDirections<volume_dim>`
- *   - `Tags::BoundaryDirectionsInterior<volume_dim>`
- *   - `interface<Tags::Mesh<volume_dim - 1>>`
- *   - `boundary<Tags::Mesh<volume_dim - 1>>`
- *
- * DataBox:
- * - Adds:
- *   - `interface<normal_dot_flux_tag>`
- *   - `boundary<normal_dot_flux_tag>`
+ *   - `fluxes`
  */
 template <typename Metavariables>
 struct InitializeFluxes {
-  static constexpr size_t volume_dim = Metavariables::system::volume_dim;
-  using normal_dot_flux_tag =
-      db::add_tag_prefix<::Tags::NormalDotFlux,
-                         typename Metavariables::system::variables_tag>;
+ private:
+  using system = typename Metavariables::system;
+  static constexpr size_t volume_dim = system::volume_dim;
+  using vars_tag = typename system::variables_tag;
+  using fluxes_tag =
+      db::add_tag_prefix<::Tags::Flux, vars_tag, tmpl::size_t<volume_dim>,
+                         Frame::Inertial>;
+  using div_fluxes_tag = db::add_tag_prefix<::Tags::div, fluxes_tag>;
+
+  template <typename Directions>
+  using face_tags =
+      tmpl::list<::Tags::Slice<Directions, fluxes_tag>,
+                 ::Tags::Slice<Directions, div_fluxes_tag>,
+                 // For the strong first-order DG scheme we also need the
+                 // interface normal dotted into the fluxes
+                 ::Tags::InterfaceComputeItem<
+                     Directions, ::Tags::ComputeNormalDotFlux<
+                                     vars_tag, volume_dim, Frame::Inertial>>>;
 
-  template <typename Tag>
-  using interface_tag =
-      Tags::Interface<Tags::InternalDirections<volume_dim>, Tag>;
-  template <typename Tag>
-  using boundary_tag =
-      Tags::Interface<Tags::BoundaryDirectionsInterior<volume_dim>, Tag>;
+  using fluxes_compute_tag =
+      elliptic::Tags::FirstOrderFluxesCompute<volume_dim, system>;
 
+  using exterior_tags = tmpl::list<
+      // On exterior (ghost) boundary faces we compute the fluxes from the
+      // data that is being set there to impose homogeneous Dirichlet boundary
+      // conditions. Then, we compute their normal-dot-fluxes. The flux
+      // divergences are sliced from the volume.
+      ::Tags::InterfaceComputeItem<
```

Comment:
Can you please briefly specify what will need to be extended to implement other BCs than homogeneous Dirichlet?

Reply:
This is actually valid for any boundary conditions, adjusted the comment. It's the actions in Elliptic/DG that need to be changed to support other boundary conditions.